cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowImpl.qll

```diff
@@ -3019,9 +3067,39 @@ private module FlowExploration {
         else result = "[" + tc.toString() + ", ... (" + len.toString() + ")]"
       )
     }
+  }
+
+  private newtype TRevPartialAccessPath =
```

Comment:
Why do we need this IPA type? It looks like the implementation is the same as `TPartialAccessPath`

Reply:
The `Nil` case is different.

Reply:
So it is. Even though the type would not be used for the backwards analysis, could we perhaps use the same IPA type to simplify things?

Reply:
Hmm, yeah I guess we could just have two different `Nil` cases. That does cut some code. I'll try it.

Reply:
Hmm, actually the cons case is different as well. It uses `Content` rather than `TypedContent`, so it's not as simple as that.

Reply:
OK, we need the two types then.